src/lib/components/layouts/DocumentBrowser.svelte

```diff
@@ -74,6 +80,26 @@
     HIDE_COUNT: width < remToPx(26),
   };
 
+  $: searchResults = documents.then((r) => excludeDeleted(r.data, $deleted));
+
+  // filter out deleted documents that haven't been purged from search yet
+  function excludeDeleted(
+    searchResults: DocumentResults,
+    deleted: Set<string>,
+  ): DocumentResults {
+    if (deleted.size === 0) return searchResults;
+
+    const filtered = searchResults.results.filter(
+      (d) => !deleted.has(String(d.id)),
+    );
+
+    return {
+      ...searchResults,
+      results: filtered,
+      count: searchResults.count - deleted.size,
```

Comment:
Should this be `filtered.length`? If there are deleted documents but they aren't present in the page, the count will mismatch the results.

Reply:
Nevermind, I forgot that count is the total # results, not the page count.